lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -2453,6 +2453,86 @@ void glow::convertPlaceholdersToConstants(Function *F,
   }
 }
 
+/// Return true if the \p node sub-tree corresponds to a scalar
+/// (Constant or Splat) and return the float value in \p retFloat.
+static bool getFloatScalar(Node *node, float *retFloat) {
+  // Iterate accross potential Tile Nodes (implied by broadcasting if any).
+  auto *n = node;
+  while (auto *TN = dyn_cast<TileNode>(n)) {
+    n = TN->getInput();
+  }
+
+  // After potential Tile nodes, it should be a singleton constant scalar node
+  // with any shape corresponding to one single element.
+  if (auto *constNode = dyn_cast<Constant>(n)) {
+    if ((constNode->getType()->getElementType() != ElemKind::FloatTy) ||
```

Comment:
We support multiple FP types, right? Float and FP16. We also have a predicate `isFPType` that can be used to check if it is one of those types.